packages/services/api/src/modules/organization/providers/organization-manager.ts

```diff
@@ -1159,14 +1362,36 @@ export class OrganizationManager {
     this.authManager.resetAccessCache();
     this.session.reset();
 
-    return {
-      ok: {
-        updatedMember: await this.getOrganizationMember({
+    const result = {
+      updatedMember: await this.getOrganizationMember({
+        organizationId: input.organizationId,
+        userId: input.userId,
+      }),
+      previousMemberRole: member.role,
+    };
+
+    if (result) {
+      this.auditLog.record(
+        {
+          eventType: 'ROLE_ASSIGNED',
+          roleAssignedAuditLogSchema: {
+            previousMemberRole: member.role ? member.role.name : null,
+            roleId: newRole.id,
+            updatedMember: member.user.email,
+            userIdAssigned: input.userId,
+          },
+        },
+        {
           organizationId: input.organizationId,
-          userId: input.userId,
-        }),
-        previousMemberRole: member.role,
-      },
+          userEmail: currentUser.email,
+          userId: currentUser.id,
+          user: currentUser,
+        },
+      );
```

Comment:
TODO @TuvalSimha:

Lets change the record method types and zod models types so the call signature looks like the following:

```ts
this.auditLog.record(
        {
          eventType: 'ROLE_ASSIGNED',
          previousMemberRole: member.role ? member.role.name : null,
          roleId: newRole.id,
          updatedMember: member.user.email,
          userIdAssigned: input.userId,
          metadata: {
            organizationId: input.organizationId,
            userEmail: currentUser.email,
            userId: currentUser.id,
            user: currentUser,
          }
        }
)
```